classes/Discount.php

```diff
@@ -224,8 +223,8 @@ public static function createOrderDiscount($order, $productList, $qtyList, $name
 		// create discount
 		$voucher = new Discount();
 		$voucher->id_discount_type = Discount::AMOUNT;
-		foreach ($languages as $language)
-			$voucher->description[$language['id_lang']] = strval($name).(int)($order->id);
+		foreach (Language::getIDs((bool)$order) as $id_lang)
```

Comment:
Which functional tests ?
The Discount class is obsolete. Do you have functionnal tests on obsolete code ? I'm surprised
I don't understand that obsolete code is optimized .....